### PR TITLE
Service Resource Fix + Snapshot Unit Test + Copyright Update

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
# Description
Service Resource Fix

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

### CODE REVIEW

**Review by shenda1 — DISMISSED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

File: examples/resources/powerflex_service/resource.tf [hunk 1]
```diff
@@ -32,3 +34,20 @@ resource "powerflex_service" "service" {
...
+
+data "powerflex_template" "template_data" {
+  template_names = ["Teample_Name"]
+}
+
+resource "powerflex_service" "service" {
+  deployment_name        = "Test-Create-U1"
+  deployment_description = "Test Service-U1"
+  template_id            = data.powerflex_template.template_data.template_details[0].id
+  firmware_id            = data.powerflex_template.template_data.template_details[0].firmware_id
```
Comment (rounak-adhikary) [inline examples/resources/powerflex_service/resource.tf:49]:
```suggestion
  template_id            = tolist(data.powerflex_template.template_data.template_details)[0].id
  firmware_id            = tolist(data.powerflex_template.template_data.template_details)[0].firmware_id
```
Set datastructure cannot be indexed directly

File: examples/resources/powerflex_service/resource.tf [hunk 2]
```diff
@@ -32,3 +34,20 @@ resource "powerflex_service" "service" {
...
+}
+
+resource "powerflex_service" "service" {
+  deployment_name        = "Test-Create-U1"
+  deployment_description = "Test Service-U1"
+  template_id            = tolist(data.powerflex_template.template_data.template_details)[0].id
+  firmware_id            = tolist(data.powerflex_template.template_data.template_details)[0].firmware_id
+  nodes                  = 0
+  clone_from_host        = "pfmc-k8s-20230809-160"
+  deployment_timeout     = 0
```
Comment (rounak-adhikary) [inline examples/resources/powerflex_service/resource.tf:52]: Also, `nodes` and `deployment_timeout` has validators. They must be at least 1 and 10 respectively.

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Review by rounak-adhikary — DISMISSED** [review]

**Review by shenda1 — APPROVED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]

**Review by gokul-srivathsan — APPROVED** [review]
LGTM

**Pull request merged**